### PR TITLE
Update Select2 4.0.2

### PR DESCRIPTION
**Pull request by SMR21**
Update Select2 4.0.2 completed. #2851

### CODE REVIEW

**Comment by kprovance** [comment]
I will need to thorough test this before committing.  :)

**Comment by Nikschavan** [comment]
Did you get a chance to review this @kprovance?

I used this in one of my projects and seems to be working just fine.

**Comment by kprovance** [comment]
No, not yet.  It's a a LOT of code and a lot of changes (and things you took out of other fields that don't use select, which I wonder why) that I have to go through for each field and test against several situations.  Plus, it's going to break compatibility with the plugin we offer  for those who don't like to use the CDN.  Evenn updating it means we're going to get complaints  about it, which means I gotta answer them all.  But mostly, the time I would have to invest in vetting such large changes would be considerable, and the truth is, no one is paying me.  We're not getting enough donations from our users (especially those who make six figures) to justify the time I have to take away from my real paying job.  I know it sucks, but it is what it is.  I'm glad it works on your fork, but I have to test against backward compatibility models, because if one thing is broken, especially with changes make to things that don't even use select2, we have tens of thousand of users that could all come back to us at once complaining about breaks...and again, they don't pay us for Redux or support, so that's more time I have to spend dealing with support.  I will get to it when I am able, I just can;t say when that will be.

File: ReduxCore/inc/fields/typography/field_typography.js [hunk 1]
```diff
@@ -349,20 +496,25 @@
...
         }
 
         // Page load. Speeds things up memory wise to offload to client
         if ( !$( '#' + mainID ).hasClass( 'typography-initialized' ) ) {
-            style = $( '#' + mainID + ' select.redux-typography-style' ).data( 'value' );
-            script = $( '#' + mainID + ' select.redux-typography-subsets' ).data( 'value' );
+            //style = $( '#' + mainID + ' select.redux-typography-style' ).data( 'value' );
+            //script = $( '#' + mainID + ' select.redux-typography-subsets' ).data( 'value' );
+            style = $( '#' + mainID + ' select.redux-typography-style' ).attr( 'data-value' );
+            script = $( '#' + mainID + ' select.redux-typography-subsets' ).attr( 'data-value' );
```
Comment (kprovance) [inline ReduxCore/inc/fields/typography/field_typography.js]: I don't get this.  What's wrong with the .data function?  No need for .attr on data attributes.

File: ReduxCore/inc/fields/typography/field_typography.js [hunk 2]
```diff
@@ -349,20 +496,25 @@
         //console.log(color);
 
         //var output = family;
+        
+        //add
+        if(active){
```
Comment (kprovance) [inline ReduxCore/inc/fields/typography/field_typography.js]: This won't work.  the google and detail vars are loaded in this if/then block, and instances of it fall out of scope when the block ends.

**Comment by kprovance** [comment]
This will be bumped to the next major version of Redux, since it needs fixing.  Thanks for the contribute!

**Pull request closed without merging**